File: src/connect4/data/Position.java
package connect4.data;

public class Position {
    public int i;
    public int j;

    public Position(int i, int j) {
        this.i = i;
        this.j = j;
    }

    public boolean isInBounds(int rows, int cols) {
        return i >= 0 && i < rows && j >= 0 && j < cols;
    }
}

File: webapp/src/classes/Position.ts
export class Position {
    public i: number;
    public j: number;

    constructor(i: number, j: number) {
        this.i = i;
        this.j = j;
    }

    public async serialize(lib: any): any {
        const PositionLib: any = await lib.connect4.data.Position;

        return await new PositionLib(this.i, this.j);
    }
}